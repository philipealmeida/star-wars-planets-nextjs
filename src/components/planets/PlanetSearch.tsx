import { ChangeEvent, useState, useEffect, useCallback } from 'react';
import styles from '@/styles/components/planets/PlanetSearch.module.css';

interface PlanetSearchProps {
  value: string;
  onChange: (value: string) => void;
}

export default function PlanetSearch({ value, onChange }: PlanetSearchProps) {
  const [localValue, setLocalValue] = useState(value);

  useEffect(() => {
    setLocalValue(value);
  }, [value]);

  const handleChange = useCallback((e: ChangeEvent<HTMLInputElement>) => {
    const newValue = e.target.value;
    setLocalValue(newValue);
    onChange(newValue);
  }, [onChange]);

  return (
    <div className={styles.search}>
      <input
        type="text"
        value={localValue}
        onChange={handleChange}
        placeholder="Search planets..."
        className={styles.input}
      />
    </div>
  );
}