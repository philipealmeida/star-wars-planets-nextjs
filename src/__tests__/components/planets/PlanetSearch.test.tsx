import { render, screen, fireEvent } from '@testing-library/react';
import PlanetSearch from '@/components/planets/PlanetSearch';

const renderComponent = (props: {
  value: string;
  onChange: (value: string) => void;
}) => {
  return render(<PlanetSearch {...props} />);
};

describe('PlanetSearch', () => {
  const mockOnChange = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('matches snapshot', () => {
    const { container } = renderComponent({
      value: '',
      onChange: mockOnChange,
    });
    expect(container).toMatchSnapshot();
  });

  it('renders with initial value', () => {
    renderComponent({
      value: 'test',
      onChange: mockOnChange,
    });
    
    const input = screen.getByPlaceholderText('Search planets...');
    expect(input).toHaveValue('test');
  });

  it('calls onChange when input value changes', () => {
    renderComponent({
      value: '',
      onChange: mockOnChange,
    });
    
    const input = screen.getByPlaceholderText('Search planets...') as HTMLInputElement;
    fireEvent.change(input, { target: { value: 'new value' } });
    
    expect(mockOnChange).toHaveBeenCalledWith('new value');
  });
});