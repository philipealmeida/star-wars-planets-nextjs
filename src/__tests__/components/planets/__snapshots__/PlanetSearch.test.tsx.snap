// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanetSearch matches snapshot 1`] = `
<div>
  <div
    class="search"
  >
    <input
      class="input"
      placeholder="Search planets..."
      type="text"
      value=""
    />
  </div>
</div>
`;
